plugins/monaco/engine.js

```diff
@@ -52,25 +52,15 @@ Text editor engine based on a Monaco instance
         // as a result of this require and some also on-the-fly (for e.g. language support)
         context.require(["vs/editor/editor.main"], function() {
             window.monacoInit.dispatchEvent(new Event("ready"));
-            
-            // Run all monaco plugins
-            const modules = $tw.modules.types["monaco"];
-            const req = Object.getOwnPropertyNames(modules);
-            if (req) {
-                if (Array.isArray(req)) {
-                    for (const r of req) {
-                        require(r);
-                    }
-                } else {
-                    require(req);
-                }
-            }
+            $tw.modules.forEachModuleOfType('monaco', function (title, exportModules) {
+                // Actually, no need to call require(title)
+            });
```

Comment:
See [boot.js in tiddlywiki](https://github.com/Jermolene/TiddlyWiki5/blob/master/boot/boot.js):

```js
$tw.modules.forEachModuleOfType = function(moduleType,callback) {
	var modules = $tw.modules.types[moduleType];
	$tw.utils.each(modules,function(element,title) {
		callback(title,$tw.modules.execute(title));
	});
};
```

this method has already executed each code of module tiddlers.

Only why CodeMirror adds these codes, I think maybe there is no auto-loading mechanism at the beginning btw, or maybe the codes themselves are unnecessary. The reason why the CodeMirror plugin does not have errors is that the CodeMirror plugin must contain at least one plugin.

I think we should do the same - using the tw-monaco plugin must be used to write tw in the first place, right?

Reply:
Sorry for the late response!

I initially tried to model it closer to CodeMirror with features being separate plugins, but then I realized it might just make it harder to install. So I ended up with the main `monaco` plugin and a `monaco-wikitext` plugin that adds wikitext-specific functionality.

We can go a little further and just merge all the `monaco-wikitext` functionality into the main `monaco` plugin as I'm not sure there is any need for it to be separate anyway. If people only want some parts of it that could be done via plugin configuration. 

What do you think?